master/internal/api_experiment.go

```diff
@@ -2337,3 +2338,109 @@ func (a *apiServer) CreateTrial(
 
 	return resp, nil
 }
+
+func (a *apiServer) PutExperimentLabel(ctx context.Context,
+	req *apiv1.PutExperimentLabelRequest,
+) (*apiv1.PutExperimentLabelResponse, error) {
+	curUser, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to get the user: %s", err)
+	}
+
+	exp, err := a.getExperiment(ctx, *curUser, int(req.ExperimentId))
+	if err != nil {
+		return nil, err
+	}
+
+	modelExp, err := model.ExperimentFromProto(exp)
+	if err != nil {
+		return nil, err
+	}
+
+	if err = exputil.AuthZProvider.Get().CanEditExperimentsMetadata(
+		ctx, *curUser, modelExp); err != nil {
+		return nil, status.Errorf(codes.PermissionDenied, err.Error())
+	}
+
+	// Labels should be unique.
+	expLabels := set.New[string]()
+	for _, label := range exp.Labels {
+		expLabels.Insert(label)
+	}
+	if expLabels.Contains(req.Label) {
+		return &apiv1.PutExperimentLabelResponse{Labels: exp.Labels}, nil
+	}
+	expLabels.Insert(req.Label)
+
+	exp.Labels = expLabels.ToSlice()
```

Comment:
Equally clear, to me
```suggestion
	if slices.Contains(exp.Labels, req.Label) {
		return &apiv1.PutExperimentLabelResponse{Labels: exp.Labels}, nil
	}
	exp.Labels = append(exp.Labels, req.Label)
```